engine/runtime/src/main/scala/org/enso/compiler/Passes.scala

```diff
@@ -114,8 +113,7 @@ class Passes(
 
   /** Configuration for the passes. */
   private val passConfig: PassConfiguration = PassConfiguration(
-    ApplicationSaturation -->> ApplicationSaturation.Configuration(),
-    AliasAnalysis         -->> AliasAnalysis.Configuration()
```

Comment:
Was the `ApplicationSaturation` pass ever actually used outside of tests??

Looking at this initializer, we can see that the `Configuration` gets initialized with an **empty** `Map` for `knownFunctions`. And if the encountered funtion is _not_ one of `knownFunctions`, it seems to always get the `Unknown` state, not influencing the codegen in any way.

I also don't see any other place where this pass is configured to _"know"_ any functions and be able to do anything useful.

So I suspect we shouldn't see _any_ difference in benchmarks, because we seem to be removing _already practically unused_ code.

Reply:
Looking at usages of the `FunctionSpec`, I cannot see _anywhere_ in the codebase where it is constructed with the `codegenHelper` different than `dummyFn` (which does not do anything). So I really suspect this pass was never really fully used.

<img width="764" alt="image" src="https://github.com/enso-org/enso/assets/1436948/a9d985e8-8757-4840-87e2-d86f485dd64f">